File: UPRP Competitions/2017/Claves.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <cctype>
#include <algorithm>
using namespace std;


int main() {
    int test, alp, num, up, low;
    string next;
    
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    cin >> test;
    
    for (int t = 1; t <= test; t++){
        cin >> next;
        alp = num = up = low = 0;

        for (int i = 0; i < next.length(); i++)
            if (isalpha(next[i])){
                alp++;
                if (isupper(next[i])) up++;
                else low++;
            }
            else if (isdigit(next[i])) num++;
                
        cout << "Caso #" << t << ": ";
        if (alp == num && up == low) cout << "Balanceada.\n";
        else if (alp == num) cout << "Parcialmente Balanceada.\n";
        else cout << "No Balanceada.\n";
    }
    
    return 0;
}

File: UPRP Competitions/2017/Claves.py
test = int(raw_input())

for i in xrange(test):
    low = up = alp = num = 0
    
    word = raw_input()
    
    for j in xrange(len(word)):
        if (word[j].isalpha()):
            alp += 1
            if (word[j].islower()): low += 1
            else: up += 1
        elif (word[j].isdigit()): num += 1
            
    if (alp == num and low == up): print("Caso #"+str(i+1)+(": Balanceada."))
    elif (alp == num): print("Caso #"+str(i+1)+(": Parcialmente Balanceada."))
    else: print("Caso #"+str(i+1)+(": No Balanceada."))
